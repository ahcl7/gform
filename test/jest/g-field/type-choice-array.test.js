import { generateComponent } from "./common";

describe('GField with type choice array', function () {
  it('should render correctly', () => {
    const component = generateComponent({
      key: 'layout2', type: 'choiceArray', label: 'Layout', choiceKey: 'type',
      fields: [{
        choiceName: 'linear', type: 'object',
        fields: [{key: 'linearName', type: 'input', flex: 'md-12'}]
      }, {
        choiceName: 'grid', type: 'object',
        fields: [{key: 'gridName', type: 'input', flex: 'md-12'}]
      }]
    });
    expect(component.html().replace(/fragment=".*?"/g, '')).toMatchSnapshot();
  });
});
