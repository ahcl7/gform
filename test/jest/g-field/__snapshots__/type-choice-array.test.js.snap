// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GField with type choice array should render correctly 1`] = `
"<div >
  <div class=\\"g-col col-md-12\\"  md12=\\"\\">
    <div z-index=\\"1000\\">
      <div class=\\"g-btn g-btn__contained g-btn__uppercase bg-blue text-white g-size__small waves-effect\\"><span class=\\"g-btn__content\\">
          ADD LAYOUT
          <i class=\\"g-icon material-icons\\" style=\\"font-size: 24px;\\">arrow_drop_down</i></span></div>
    </div>
  </div>
</div>"
`;
